src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.tsx

```diff
@@ -1,140 +1,117 @@
 import withToggleVisibilityView from '@components/withToggleVisibilityView';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
 import usePrevious from '@hooks/usePrevious';
 import useStyleUtils from '@hooks/useStyleUtils';
-import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import canFocusInputOnScreenFocus from '@libs/canFocusInputOnScreenFocus';
-import compose from '@libs/compose';
 import * as ErrorUtils from '@libs/ErrorUtils';
 import * as ValidationUtils from '@libs/ValidationUtils';
 import ChangeExpensifyLoginLink from '@pages/signin/ChangeExpensifyLoginLink';
 import Terms from '@pages/signin/Terms';
-import * as Session from '@userActions/Session';
+import * as SessionActions from '@userActions/Session';
 import * as User from '@userActions/User';
 import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type {Account, Credentials, Session} from '@src/types/onyx';
+import type {Errors} from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
-const propTypes = {
-    /* Onyx Props */
-
+type BaseValidateCodeFormOnyxProps = {
     /** The details about the account that the user is signing in with */
-    account: PropTypes.shape({
-        /** Whether or not two-factor authentication is required */
-        requiresTwoFactorAuth: PropTypes.bool,
-
-        /** Whether or not a sign on form is loading (being submitted) */
-        isLoading: PropTypes.bool,
-
-        /** Whether or not the user has SAML enabled on their account */
-        isSAMLEnabled: PropTypes.bool,
-
-        /** Whether or not SAML is required on the account */
-        isSAMLRequired: PropTypes.bool,
-    }),
-
-    /** The credentials of the person signing in */
-    credentials: PropTypes.shape({
-        /** The login of the person signing in */
-        login: PropTypes.string,
-    }),
-
-    /** Session of currently logged in user */
-    session: PropTypes.shape({
-        /** Currently logged in user authToken */
-        authToken: PropTypes.string,
-    }),
+    account: OnyxEntry<Account>;
 
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
+    /** The credentials of the person logging in */
+    credentials: OnyxEntry<Credentials>;
 
-    /** Specifies autocomplete hints for the system, so it can provide autofill */
-    autoComplete: PropTypes.oneOf(['sms-otp', 'one-time-code']).isRequired,
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
 
-    /** Determines if user is switched to using recovery code instead of 2fa code */
-    isUsingRecoveryCode: PropTypes.bool.isRequired,
+type BaseValidateCodeFormProps = WithToggleVisibilityViewProps &
+    BaseValidateCodeFormOnyxProps & {
```

Comment:
You can reuse `ValidateCodeFormProps` here as well instead of writing `isUsingRecoveryCode` and `setIsUsingRecoveryCode` again
```suggestion
    BaseValidateCodeFormOnyxProps &  ValidateCodeFormProps & {
```